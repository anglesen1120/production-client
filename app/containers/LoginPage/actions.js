/*
 *
 * LoginPage actions
 *
 */

import {
  CHANGE_LOGIN,
  CHANGE_PASSWORD,
  TOGGLE_MASTER,
  TOGGLE_PASSWORD,
  ENTER_LOGIN,
  ENTER_LOGIN_SUCCESS,
  ENTER_LOGIN_ERROR,
} from './constants';

export function changeLoginAction(login) {
  return {
    type: CHANGE_LOGIN,
    login,
  };
}

export function changePasswordAction(password) {
  return {
    type: CHANGE_PASSWORD,
    password,
  };
}

export function toggleMasterAction() {
  return {
    type: TOGGLE_MASTER,
  };
}

export function togglePasswordAction() {
  return {
    type: TOGGLE_PASSWORD,
  };
}

export function enterLoginAction() {
  return {
    type: ENTER_LOGIN,
  };
}

export function enterLoginSuccessAction(token) {
  return {
    type: ENTER_LOGIN_SUCCESS,
    token,
  };
}

export function enterLoginErrorAction(error) {
  return {
    type: ENTER_LOGIN_ERROR,
    error,
  };
}
