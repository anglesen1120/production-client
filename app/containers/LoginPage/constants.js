/*
 *
 * LoginPage constants
 *
 */

export const CHANGE_LOGIN = 'app/LoginPage/CHANGE_LOGIN';
export const CHANGE_PASSWORD = 'app/LoginPage/CHANGE_PASSWORD';
export const TOGGLE_MASTER = 'app/LoginPage/TOGGLE_MASTER';
export const TOGGLE_PASSWORD = 'app/LoginPage/TOGGLE_PASSWORD';
export const ENTER_LOGIN = 'app/LoginPage/ENTER_LOGIN';
export const ENTER_LOGIN_SUCCESS = 'app/LoginPage/ENTER_LOGIN_SUCCESS';
export const ENTER_LOGIN_ERROR = 'app/LoginPage/ENTER_LOGIN_ERROR';
