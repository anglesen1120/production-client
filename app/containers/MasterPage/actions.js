/*
 *
 * MasterPage actions
 *
 */

import {
  GET_CUSTOMERS,
  GET_CUSTOMERS_SUCCESS,
  GET_CUSTOMERS_ERROR,
  CREATE_CUSTOMER,
  CREATE_CUSTOMER_SUCCESS,
  CREATE_CUSTOMER_ERROR,
  GET_PRODUCTION_MACHINES,
  GET_PRODUCTION_MACHINES_SUCCESS,
  GET_PRODUCTION_MACHINES_ERROR,
  CREATE_PRODUCTION_MACHINE,
  CREATE_PRODUCTION_MACHINE_SUCCESS,
  CREATE_PRODUCTION_MACHINE_ERROR,
  GET_PRODUCTION_TASKS,
  GET_PRODUCTION_TASKS_SUCCESS,
  GET_PRODUCTION_TASKS_ERROR,
  CREATE_PRODUCTION_TASK,
  CREATE_PRODUCTION_TASK_SUCCESS,
  CREATE_PRODUCTION_TASK_ERROR,
} from './constants';

export function getCustomersAction() {
  return {
    type: GET_CUSTOMERS,
  };
}

export function getCustomersSuccessAction(customers) {
  return {
    type: GET_CUSTOMERS_SUCCESS,
    customers,
  };
}

export function getCustomersErrorAction(error) {
  return {
    type: GET_CUSTOMERS_ERROR,
    error,
  };
}

export function createCustomerAction() {
  return {
    type: CREATE_CUSTOMER,
  };
}

export function createCustomerSuccessAction() {
  return {
    type: CREATE_CUSTOMER_SUCCESS,
  };
}

export function createCustomerErrorAction(error) {
  return {
    type: CREATE_CUSTOMER_ERROR,
    error,
  };
}

export function getProductionMachinesAction() {
  return {
    type: GET_PRODUCTION_MACHINES,
  };
}

export function getProductionMachinesSuccessAction(productionMachines) {
  return {
    type: GET_PRODUCTION_MACHINES_SUCCESS,
    productionMachines,
  };
}

export function getProductionMachinesErrorAction(error) {
  return {
    type: GET_PRODUCTION_MACHINES_ERROR,
    error,
  };
}

export function createProductionMachineAction() {
  return {
    type: CREATE_PRODUCTION_MACHINE,
  };
}

export function createProductionMachineSuccessAction() {
  return {
    type: CREATE_PRODUCTION_MACHINE_SUCCESS,
  };
}

export function createProductionMachineErrorAction(error) {
  return {
    type: CREATE_PRODUCTION_MACHINE_ERROR,
    error,
  };
}

export function getProductionTasksAction() {
  return {
    type: GET_PRODUCTION_TASKS,
  };
}

export function getProductionTasksSuccessAction(productionTasks) {
  return {
    type: GET_PRODUCTION_TASKS_SUCCESS,
    productionTasks,
  };
}

export function getProductionTasksErrorAction(error) {
  return {
    type: GET_PRODUCTION_TASKS_ERROR,
    error,
  };
}

export function createProductionTaskAction() {
  return {
    type: CREATE_PRODUCTION_TASK,
  };
}

export function createProductionTaskSuccessAction() {
  return {
    type: CREATE_PRODUCTION_TASK_SUCCESS,
  };
}

export function createProductionTaskErrorAction(error) {
  return {
    type: CREATE_PRODUCTION_TASK_ERROR,
    error,
  };
}
