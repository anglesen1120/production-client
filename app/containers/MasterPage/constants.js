/*
 *
 * MasterPage constants
 *
 */

export const DEFAULT_ACTION = 'app/MasterPage/DEFAULT_ACTION';
