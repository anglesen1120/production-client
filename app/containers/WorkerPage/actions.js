/*
 *
 * WorkerPage actions
 *
 */

import socketIOClient from 'socket.io-client';
import {
  GET_TASK,
  GET_TASK_SUCCESS,
  GET_TASK_ERROR,
  UPDATE_TASK,
  UPDATE_TASK_SUCCESS,
  UPDATE_TASK_ERROR,
  CHANGE_COUNTDOWN_TIME,
  CHANGE_SCHEDULED_QUANTITY,
  CHANGE_MADE_QUANTITY,
  PAUSE_COUNTDOWN_TIME,
} from './constants';

const socket = socketIOClient('http://localhost:3001/production');

export function getTaskAction() {
  return {
    type: GET_TASK,
  };
}

export function getTaskSuccessAction(task) {
  socket.emit('login', { task });

  return {
    type: GET_TASK_SUCCESS,
    task,
  };
}

export function getTaskErrorAction(error) {
  return {
    type: GET_TASK_ERROR,
    error,
  };
}

export function updateTaskAction() {
  return {
    type: UPDATE_TASK,
  };
}

export function updateTaskSuccessAction(task) {
  socket.emit('login', { task });

  return {
    type: UPDATE_TASK_SUCCESS,
    task,
  };
}

export function updateTaskErrorAction(error) {
  return {
    type: UPDATE_TASK_ERROR,
    error,
  };
}

export function changeScheduledQuantityAction() {
  return {
    type: CHANGE_SCHEDULED_QUANTITY,
  };
}

export function changeMadeQuantityAction() {
  return {
    type: CHANGE_MADE_QUANTITY,
  };
}

export function changeCountdownTimeAction() {
  return {
    type: CHANGE_COUNTDOWN_TIME,
  };
}

export function pauseCountdownTimeAction() {
  return {
    type: PAUSE_COUNTDOWN_TIME,
  };
}
