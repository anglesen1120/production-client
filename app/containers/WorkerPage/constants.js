/*
 *
 * WorkerPage constants
 *
 */

export const GET_TASK = 'app/WorkerPage/GET_TASK';
export const GET_TASK_SUCCESS = 'app/WorkerPage/GET_TASK_SUCCESS';
export const GET_TASK_ERROR = 'app/WorkerPage/GET_TASK_ERROR';

export const UPDATE_TASK = 'app/WorkerPage/UPDATE_TASK';
export const UPDATE_TASK_SUCCESS = 'app/WorkerPage/UPDATE_TASK_SUCCESS';
export const UPDATE_TASK_ERROR = 'app/WorkerPage/UPDATE_TASK_ERROR';
