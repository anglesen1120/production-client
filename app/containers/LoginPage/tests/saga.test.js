import { put, takeLatest } from 'redux-saga/effects';
import loginPageSaga, { enterLogin } from '../saga';
import { ENTER_LOGIN } from '../constants';
import { enterLoginSuccessAction, enterLoginErrorAction } from '../actions';

const login = 'test';
const password = 'test2';

describe('enterLogin Saga', () => {
  let enterLoginGenerator;

  beforeEach(() => {
    enterLoginGenerator = enterLogin();

    const selectDescriptor = enterLoginGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = enterLoginGenerator.next(login, password).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the enterLoginSuccessAction action if it requests the data successfully', () => {
    const response = { token: { token: 'test' } };
    const putDescriptor = enterLoginGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(enterLoginSuccessAction(response)));
  });

  it('should call the enterLoginErrorAction action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = enterLoginGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(enterLoginErrorAction(response)));
  });
});

describe('loginPageSaga Saga', () => {
  const loginPage = loginPageSaga();

  it('should start task to watch for ENTER_LOGIN action', () => {
    const takeLatestDescriptor = loginPage.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(ENTER_LOGIN, loginPage));
  });
});
