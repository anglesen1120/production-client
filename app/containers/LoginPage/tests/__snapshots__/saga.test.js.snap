// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enterLogin Saga should call the enterLoginErrorAction action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`enterLogin Saga should call the enterLoginErrorAction action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`enterLogin Saga should dispatch the enterLoginSuccessAction action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`enterLogin Saga should dispatch the enterLoginSuccessAction action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;
